backend-service/app/models/kafka/MetastoreAuditProcessor.java

```diff
@@ -74,19 +76,18 @@ else if (record.get("metastorePartitionAuditContent") != null) {
       final GenericData.Record rec = newInfo != null ? (GenericData.Record) newInfo : (GenericData.Record) oldInfo;
       final String dbName = String.valueOf(rec.get("dbName"));
       final String tableName = String.valueOf(rec.get("tableName"));
-      final String partition = String.valueOf(rec.get("values"));
-      final String location = String.valueOf(rec.get("location"));
-      final String owner = String.valueOf(rec.get("owner"));
+      // set null / "null" partition to '?' for primary key
+      final String partition = StringUtil.toStringReplaceNull(rec.get("values"), "?");
+      final String location = StringUtil.toStringReplaceNull(rec.get("location"), null);
+      final String owner = StringUtil.toStringReplaceNull(rec.get("owner"), null);
       final long createTime = (long) rec.get("createTime");
       final long lastAccessTime = (long) rec.get("lastAccessTime");
 
       eventRecord = new MetastoreAuditRecord(server, instance, appName, eventName, eventType, timestamp);
       eventRecord.setEventInfo(metastoreThriftUri, metastoreVersion, isSuccessful, isDataDeleted);
-      // set null partition to '?' for primary key
-      eventRecord.setTableInfo(dbName, tableName, (partition != null ? partition : "?"),
-          location, owner, createTime, lastAccessTime);
-      eventRecord.setOldInfo(String.valueOf(oldInfo));
-      eventRecord.setNewInfo(String.valueOf(newInfo));
+      eventRecord.setTableInfo(dbName, tableName, partition, location, owner, createTime, lastAccessTime);
+      eventRecord.setOldInfo(StringUtil.toStringReplaceNull(oldInfo, null));
```

Comment:
For storage footprint consideration, we might need to ignore the "old_info", and also trim down the detail of "new_info".

"new_info" can be 4KB~11KB each with very verbose Hive table metadata.